src/component/isolation.rs

```diff
@@ -42,6 +52,18 @@ impl IsolatedProcessWrapper {
 
         resp
     }
+
+    // The `heartbeat` function is called periodically
+    pub fn heartbeat(&mut self) {
+        if self.process_handle.is_none() {
+            return;
+        }
+
+        if Instant::now() - self.last_accessed > EXPIRY_DURATION {
+            debug!("Shutting down unused function {:?}", self.process_handle);
+            self.process_handle = None
```

Comment:
How is this enough to actually shut down the process?

Reply:
The Drop implementation does that. Doing it this way is nice, since it handles ‘deactivate’ as well. Also you’d always want to shut down a dropped process.

(Google “rust destructor” for more info)

Reply:
The Drop implementation does that. Doing it this way is nice, since it handles ‘deactivate’ as well. Also you’d always want to shut down a dropped process.

(Google “rust destructor” for more info)